Upgrade to Electron 13

TheCleric opened the pull request:
This PR upgrades Nativefier to Electron 13. It will stay in draft until we feel its well tested.

Major Version Checklist

- [X] Wait a few weeks to let it stabilize. Never upgrade Nativefier to a .0.0.
- [x] Thoroughly digest the new version's breaking changes, grepping our codebase for every changed API. If called for by the breaking changes, perform the necessary API changes
- [X] Bump src/constants.ts / DEFAULT_ELECTRON_VERSION & DEFAULT_CHROME_VERSION and app / package.json / devDeps / electron
- [x] On Windows, macOS, Linux, test for regression and crashes:
    - [x] With npm test and npm run test:manual
        - [x] MacOS
        - [x] Windows
        - [x] Linux (@ronjouch can you handle this?)
    - [x] With extra manual testing
        - [x] MacOS
        - [x] Windows
        - [x] Linux (@ronjouch can you handle this?)

TheCleric commented:
Actually this will continue to be an issue on Mac. I realized in my test that I had a user agent override that gave me a false success. Without a user agent override, v13 as of 13.1.2 still crashes on load in Mac.

See

https://github.com/electron/electron/issues/29343

ronjouch commented:
@TheCleric 

> This PR upgrades Nativefier to Electron 13. It will stay in draft until we feel its well tested.

👍!

> Let me know how else we'd like to attack this

I dumped my usual workflow at https://github.com/nativefier/nativefier/blob/master/HACKING.md#major-updating-electron

> Actually this will continue to be an issue on Mac. I realized in my test that I had a user agent override that gave me a false success. Without a user agent override, v13 as of 13.1.2 still crashes on load in Mac. See https://github.com/electron/electron/issues/29343

Alrighty, let's keep this PR as draft until this is fixed.

ronjouch commented:
@TheCleric wheeeee, I see that https://github.com/electron/electron/releases/tag/v13.1.5 was released, featuring:

> Fixed crashes on macOS when Geolocation was used. [#29912](https://github.com/electron/electron/pull/29912)

TheCleric commented:
Breaking changes checklist:

- [X] API Changed: session.setPermissionCheckHandler(handler)
    - NOTES: Not used
- [X] Removed: shell.moveItemToTrash()
    - NOTES: Not used
- [X] Removed: BrowserWindow.addExtension(path)
    - NOTES: Not used
- [X] Removed: BrowserWindow.addDevToolsExtension(path)
    - NOTES: Not used
- [X] Removed: BrowserWindow.removeExtension(name)
    - NOTES: Not used
- [X] Removed: BrowserWindow.removeDevToolsExtension(name)
    - NOTES: Not used
- [X] Removed: BrowserWindow.getExtensions()
    - NOTES: Not used
- [X] Removed: BrowserWindow.getDevToolsExtensions()
    - NOTES: Not used
- [X] Removed: systemPreferences.isDarkMode()
    - NOTES: Not used
- [X] Removed: systemPreferences.isInvertedColorScheme()
    - NOTES: Not used
- [X] Removed: systemPreferences.isHighContrastColorScheme()
    - NOTES: Not used
- [X] Deprecated: WebContents new-window event
    -  NOTES: We are still using this, and should migrate off when we can, but for now new-window provides functionality that the new setWindowOpenHandler does not yet support.

ronjouch commented:
Wheee. Will test under Linux soon and report here.

ronjouch commented on src/constants.ts:
```diff
@@ -4,18 +4,18 @@ export const DEFAULT_APP_NAME = 'APP';
 // Update each of these periodically
 // https://product-details.mozilla.org/1.0/firefox_versions.json
-export const DEFAULT_FIREFOX_VERSION = '89.0';
+export const DEFAULT_FIREFOX_VERSION = '89.0.2';
```
```suggestion
export const DEFAULT_FIREFOX_VERSION = '90.0';
```

ronjouch reviewed (approved):
:tada: , `test:manual` is all good on Linux!

I also:

- Fixed two warnings about now-useless DT types, see https://github.com/nativefier/nativefier/commit/5d9cb9173997b296db7bc056405a16aa4f48f129
- Merged master into your branch
- Bumped the default Firefox version

Feel free to merge this whenever you're done with Windows tests; I'll cut a release

TheCleric commented:
@ronjouch everything seems good here!

ronjouch commented:
@TheCleric 👍, released in https://github.com/nativefier/nativefier/releases/tag/v45.0.0 , now on npm.
